### PR TITLE
python3-gitlab: update to 3.6.0 and adopt

### PR DESCRIPTION
**Pull request by ishaanbhimwal**
#### Testing the changes
- I tested the changes in this PR: **briefly**

### CODE REVIEW

File: srcpkgs/python3-gitlab/template [hunk 1]
```diff
@@ -1,21 +1,18 @@
...
 pkgname=python3-gitlab
-version=1.12.1
-revision=5
+version=3.6.0
+revision=1
 wrksrc=python-gitlab-$version
 build_style=python3-module
 hostmakedepends="python3-setuptools"
-depends="python3-setuptools python3-requests python3-six"
+depends="python3-setuptools python3-requests python3-requests-toolbelt python3-six"
```
Comment (paper42) [inline srcpkgs/python3-gitlab/template:8]: why is setuptools required? six is not mentioned at all in upstream.

File: srcpkgs/python3-gitlab/template [hunk 2]
```diff
@@ -1,21 +1,18 @@
...
 changelog="https://raw.githubusercontent.com/python-gitlab/python-gitlab/master/ChangeLog.rst"
 distfiles="${PYPI_SITE}/p/python-gitlab/python-gitlab-${version}.tar.gz"
-checksum=984e110c1f76fd939652c30ce3101267a7064e34417cbfc4687e6106d4db54ec
+checksum=901c54ff926f10479cb591a34d65f0a3022f2bcc41074f9a192c7fa7e4c57061
 conflicts="python-gitlab>=0"
-
-do_check() {
-	python2 setup.py test
-}
+make_check=ci-skip #fails in do_check() due to "gitlab.exceptions.GitlabGetError: 404: 404 Project Not Found"
```
Comment (paper42) [inline srcpkgs/python3-gitlab/template:18]: This failure is also happening on my computer, so it is not CI specific

Comment (paper42) [reply srcpkgs/python3-gitlab/template:18]: run tests with pytest and disable functional tests
```
checkdepends="python3-pytest $depends"
post_extract() {
	rm -r tests/functional
}
```
there might be some more missing dependencies from checkdepends

File: srcpkgs/python3-gitlab/template (hunk 1 above)
Comment (ishaanbhimwal) [reply srcpkgs/python3-gitlab/template]: they were already there so i didn't consider removing them. now that i have tested without them, the package works just fine.

**Comment by ishaanbhimwal** [comment]
ping @paper42

**Pull request merged**